### ISSUE TITLE
ScriptApiItem::getItemCallback does not accept __call-able tables

### ISSUE THREAD

**Issue by t4im** [minetest/minetest#3161]
s_item.cpp's [ScriptApiItem::getItemCallback](https://github.com/minetest/minetest/blob/3304e1e517fb8aac008c4684e72a4b59b408414a/src/script/cpp_api/s_item.cpp#L212)
doesn't allow to set tables with `__call` metamethod as item callbacks.
This seems to be the case in a [few other places](https://github.com/minetest/minetest/search?utf8=%E2%9C%93&q=lua_type%28L%2C+-1%29+%3D%3D+LUA_TFUNCTION) as well.

Besides generally limiting options, it especially prevents wrapping callbacks with stubs/spies/mocks for testing with [cubictest](https://github.com/t4im/shakedown#cubictest).

This blocks for example the [coretest](https://github.com/t4im/coretest) provided in #3160, being properly run without creating false positives.

**Comment by t4im** [comment]
This seems to be the case in a [few other cases](https://github.com/minetest/minetest/search?utf8=%E2%9C%93&q=lua_type%28L%2C+-1%29+%3D%3D+LUA_TFUNCTION) as well.

**Comment by t4im** [comment]
Not sure why you consider this unconfirmed. The code seems still to be the same: https://github.com/minetest/minetest/blob/325bf680410e8012394e5f3ba5ba947c69034899/src/script/cpp_api/s_item.cpp#L245-L253

I saw you are interested in improving lua-testability in minetest, @rubenwardy. Which makes me quite happy.
No matter what solution you are settling on; something that adds stubs spies or mocks will likely pass a callable table to keep track of how and how often it was called.

Being able to attach metadata to callbacks might also have come in handy for profiling (although that was solved differently now)

I do understand however that the effort is too much just to allow for more flexible testing at this time, when even the most basic testing is barely available. Thus closing this as a low priority feature request, possibly to be revisited when lua testing has progressed.
